### PR TITLE
Chat entity

### LINKED ISSUE #22: Chat Service
Making chat room intended for communication between two users.And make repository , service for it and contoller

### PR DESCRIPTION
**Pull request by SerhiiPoliakov1**

### CODE REVIEW

**Comment by bsolomko** [comment]
pls format

File: src/main/java/co/norse/chat/controller/ChatController.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        Chat chat = new Chat(id, recipientId);
+        chatService.addChat(chat);
+        return chat;
+    }
+
+    @PostMapping("/user/{id}/chat/{chatId}/message/send/{recipientId}")
+    public Message sendMessage(@PathVariable Long id, @PathVariable Long chatId, @PathVariable Long recipientId, @RequestBody Message message) throws Exception {
+
+        if(chatService.getChatById(chatId) == null) throw new Exception("Error chat ID");
+        Message _message = new Message();
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/controller/ChatController.java]: rename _message to messageModel

File: src/main/java/co/norse/chat/controller/ChatController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    private ChatService chatService;
+
+    @PostMapping("/user/{id}/chat/add/{recipientId}")
+    public Chat createChat(@PathVariable Long id, @PathVariable Long recipientId) {
+        Chat chat = new Chat(id, recipientId);
+        chatService.addChat(chat);
+        return chat;
+    }
+
+    @PostMapping("/user/{id}/chat/{chatId}/message/send/{recipientId}")
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/controller/ChatController.java]: i'd rather redisign it /char/{chatId} and pass all params as requestBody

File: src/main/java/co/norse/chat/controller/ChatController.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        if(chatService.getChatById(chatId) == null) throw new Exception("Error chat ID");
+        Message _message = new Message();
+        _message.setMessage(message.getMessage());
+        _message.setSenderId(id);
+        _message.setRecipientID(recipientId);
+        messageService.addMessage(_message);
+        chatService.addMessageInChat(chatId, _message);
+        return _message;
+    }
+    @GetMapping("/user/{id}/chat/{chatId}/message")
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/controller/ChatController.java]: same here

File: src/main/java/co/norse/chat/controller/ChatController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    public List<Message> getChat(@PathVariable Long id, @PathVariable Long chatId){
+    Chat chat =  chatService.getChatById(chatId);
+    List<Message> messages = chat.getMessages();
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/controller/ChatController.java]: inline

File: src/main/java/co/norse/chat/model/Chat.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
+package co.norse.chat.model;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class Chat {
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/model/Chat.java:8]: hashcode equals etc

File: src/main/java/co/norse/chat/repository/ChatRepository.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    public void deleteChatByID(Long id) {
+        this.chatStore = chatStore.stream().filter(chat -> chat.getId() != id).collect(Collectors.toList());
+    }
+
+    public void deleteAllChat(){
+        this.chatStore.clear();
+    }
+
+    public List<Chat> getAllChat(){
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/repository/ChatRepository.java]: getAllChats

File: src/main/java/co/norse/chat/repository/ChatRepository.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        this.chatStore = new ArrayList<>();
+    }
+
+
+    public void addChat(Chat chat) {
+        chatStore.add(chat);
+    }
+
+    public void deleteChatByID(Long id) {
+        this.chatStore = chatStore.stream().filter(chat -> chat.getId() != id).collect(Collectors.toList());
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/repository/ChatRepository.java:24]: remove this

File: src/main/java/co/norse/chat/repository/ChatRepository.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    public void deleteAllChat(){
+        this.chatStore.clear();
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/repository/ChatRepository.java:28]: same here

File: src/main/java/co/norse/chat/repository/ChatRepository.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+    public void addChat(Chat chat) {
+        chatStore.add(chat);
+    }
+
+    public void deleteChatByID(Long id) {
+        this.chatStore = chatStore.stream().filter(chat -> chat.getId() != id).collect(Collectors.toList());
+    }
+
+    public void deleteAllChat(){
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/repository/ChatRepository.java]: deleteAllChats

File: src/main/java/co/norse/chat/repository/ChatRepository.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        this.chatStore.clear();
+    }
+
+    public List<Chat> getAllChat(){
+        return chatStore;
+    }
+
+    public Chat getChatByID(long id){
```
Comment (bsolomko) [inline src/main/java/co/norse/chat/repository/ChatRepository.java]: getChatById

**Review by bsolomko — CHANGES_REQUESTED** [review]

File: src/main/java/co/norse/chat/model/Chat.java (hunk 5 above)
Comment (SerhiiPoliakov1) [reply src/main/java/co/norse/chat/model/Chat.java:8]: accept

**Review by SerhiiPoliakov1 — COMMENTED** [review]
accept

**Review by bsolomko — APPROVED** [review]

**Pull request merged**